src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java

```diff
@@ -371,7 +389,19 @@ public void onDeath(@NotNull final DamageSource damageSource)
             {
                 for (final Player player : PermissionUtils.getPlayersWithAtLeastRank(citizenColonyHandler.getColony(), Rank.OFFICER))
                 {
-                    if (player.getID().equals(damageSource.getTrueSource().getUniqueID()))
+                    boolean isBarbarianClose = true;
+                    for(final AbstractEntityBarbarian barbarian : this.getCitizenColonyHandler().getColony().getBarbManager().getHorde()) {
+                        final EntityCitizen citizen = new EntityCitizen(this.getEntityWorld());
+                        if(MathUtils.twoDimDistance(barbarian.getPosition(), citizen.getPosition()) < BARB_DISTANCE_FOR_FREE_DEATH)
```

Comment:
does the 2dim distance return a square number? Then we need to make Barb_distance ^ 2

Reply:
does not